### PR TITLE
focus: emit `request::activate` signal

### PR DESCRIPTION
**Pull request by Ram-Z**

### CODE REVIEW

File: lib/awful/menu.lua.in [hunk 1]
```diff
@@ -471,6 +471,7 @@ function menu.clients(args, item_args)
                     tags.viewmore(c:tags(), c.screen)
                 end
                 capi.client.focus = c
```
Comment (Elv13) [inline lib/awful/menu.lua.in]: I think you forgot to remove this line

File: lib/awful/client.lua.in [hunk 2]
```diff
@@ -208,6 +208,7 @@ function client.focus.history.previous()
     end
     local c = client.focus.history.get(s, 1)
     if c then capi.client.focus = c end
```
Comment (Elv13) [inline lib/awful/client.lua.in]: Please put the emit in the if and remove the old capi.client.focus, otherwise this will explode

**Comment by Ram-Z** [comment]
Wow... This was terrible... Had to repeat the same thing 9 times and managed to mess it up twice. :/

All fixed now.

File: lib/awful/menu.lua.in [hunk 3]
```diff
@@ -470,7 +470,7 @@ function menu.clients(args, item_args)
                 if not c:isvisible() then
                     tags.viewmore(c:tags(), c.screen)
                 end
-                capi.client.focus = c
+                c:emit_signal("request::activate", "focus")
```
Comment (Elv13) [inline lib/awful/menu.lua.in]: sorry to have missed this in my previous review, but I think "menu.clients" would be more descriptive. Other than that this PR look good. Once you fix this, if nobody complain I will go ahead and merge it.

File: lib/awful/client.lua.in [hunk 4]
```diff
@@ -71,7 +71,7 @@ function client.jumpto(c, merge)
     end
 
     -- focus the client
-    capi.client.focus = c
+    c:emit_signal("request::activate", "client.jumpto")
     c:raise()
```
Comment (blueyed) [inline lib/awful/client.lua.in]: The `c::raise` should be removed.  That's what the signal handler is supposed to to - see `ewmh.activate`.

**Comment by blueyed** [comment]
I think this should emit `request::focus` instead, given that it only sets the focus, but it not meant to raise (always).

**Comment by blueyed** [comment]
@Ram-Z 
What do you think about my previous comment?

If you agree, but don't like to update the PR, I'd be happy to pick it up.

See also https://github.com/awesomeWM/awesome/issues/217.

**Comment by Ram-Z** [comment]
On May 06, 2015 at 14:12, Daniel Hahler wrote:
> What do you think about my previous comment?
Makes sense.

> If you agree, but don't like to update the PR, I'd be happy to pick it up.
Yes, please do, I don't have much time right now. Thanks

**Comment by blueyed** [comment]
Followed up in #224.

Thanks for the initial PR and idea!

**Pull request closed without merging**